accounting/transferwise/models.py

```diff
@@ -0,0 +1,179 @@
+        return [rate.provider for rate in hourly_rates]
+
+    @property
+    def csv_filename(self):
+        """
+        Return a CSV filename for this bulk payment.
+        """
+        return 'transferwise_bulk_payment_csv_{sender}_{date}.csv'.format(
+            sender=self.sender.user.username,
+            date=self.date.strftime("%Y-%m-%d"),
+        )
+
+    def create_payments(self):
+        """
+        Create the individual TransferWise payments needed for this bulk payment.
+
+        The chosen invoice is the one that matches this bulk payment's date's month for the given payment's
+        sender and recipient. If an invoice does not fit this criteria, it gets the latest one between the two.
+
+        Note that the invoice is the source of truth for payments, and thus should exist before making a payment.
+
+        TODO: Besides creating an internal version of the payment for tracking certain data (i.e. recipient ID,
+        TODO: invoice, etc.), this should eventually actually call the API to create real transfers. At that stage,
+        TODO: we could then skip generating CSVs and create all the transfers in one go, and call them a 'bulk payment',
+        TODO: even if that terminology is only to represent how we're doing it internally.
+        """
+        for recipient in self.recipients:
+            try:
+                # Get an invoice matching this bulk payment's month & year.
+                invoice = recipient.provider_invoices.filter(
+                    client=self.sender,
+                    date__year=self.date.year,
+                    date__month=self.date.month
+                ).latest('date')
+            except Invoice.DoesNotExist:
+                # Fallback to the latest one, or just skip the recipient.
+                try:
+                    invoice = recipient.provider_invoices.filter(client=self.sender).latest('date')
+                except Invoice.DoesNotExist:
+                    continue
+
+            try:
+                self.payments.create(bulk_payment=self, invoice=invoice)
+            except (IntegrityError, ValidationError):
+                continue
+
+    def save(self, *args, **kwargs):
+        """
+        Convert the bulk payment to a CSV and/or upload it to Google Drive after a successful save.
+        """
+        super().save(**kwargs)
+        if self.auto_create_payments_on_save:
+            self.create_payments()
+        if self.auto_create_csv_on_save:
+            self.to_bulk_payment_csv(self.payments.all())
+        if self.auto_upload_google_drive_on_save:
+            file = self.upload_to_google_drive(
+                file_path=self.csv_path,
+                target_path=[self.date.strftime('%Y'), 'invoices-in', self.date.strftime('%m')],
+                title=self.csv_filename,
+            )
+            self.csv_path = file['webContentLink']
+            TransferWiseBulkPayment.objects.filter(pk=self.pk).update(csv_path=self.csv_path)
```

Comment:
@UmanShahzad Same comment here about creating a separate method for the logic that updates `self.csv_path`.

Reply:
Yep, see my comment above and let me know what you think about it.